src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java

```diff
@@ -127,9 +129,26 @@ public void setVaultAccessor(VaultAccessor vaultAccessor) {
             throw new VaultPluginException("The vault url was not configured - please specify the vault url to use.");
         }
 
+        VaultConfig vaultConfig = null;
+        try {
+            vaultConfig = new VaultConfig()
+                    .address(configuration.getVaultUrl())
+                    .sslConfig(new SslConfig().verify(configuration.isSkipSslVerification()).build());
+
+            if (configuration.getVaultNamespace() != null && !configuration.getVaultNamespace().isEmpty()) {
+                    vaultConfig.nameSpace(configuration.getVaultNamespace());
+            }
+        } catch (VaultException e) {
+            throw new VaultPluginException("Could not set up VaultConfig.", e);
+        }
         VaultCredential credential = retrieveVaultCredentials(build);
 
-        vaultAccessor.init(url, credential, configuration.isSkipSslVerification());
+        vaultAccessor.setConfig(vaultConfig);
```

Comment:
This could be a builder pattern

Reply:
Sorry disregard the above comment, I did not see the full context when looking at the code 😆